src/rules/banDomGlobalRule.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright 2014 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+
+import * as Lint from "../index";
+
+export class Rule extends Lint.Rules.TypedRule {
+    /* tslint:disable:object-literal-sort-keys */
+    public static metadata: Lint.IRuleMetadata = {
```

Comment:
Could you add a rationale explaining why this rule is useful? It would be confusing as a newcomer to TSLint reading the rule page on palantir.github.io/tslint why this would be useful.

It should probably also recommend specifically not using the dom lib in your tsconfig if you're not writing a browser app.

Reply:
Haven't you check out [New rule to forbid access to commonly-confused ambient globals](https://github.com/palantir/tslint/issues/922)? I write browser app primarily and have been bit by `name` in `lib.dom.d.ts`. Maybe I should rename this to `forbid-ambient-globals` and provides options like 
```json
[
  {
    "file": "lib.dom.d.ts",
    "variables": [
      "event",
      "name"
    ]
  },
  {
    "file": "@types/jquery/index.d.ts",
    "variables": [
      "$"
    ]
  }
]
```

Reply:
Definition file names are redundant since we cannot use a global declared in lib A if it is also declared and used in lib B.